### ISSUE TITLE
signalsReady doesn't work on registerSingletonWithDependencies

### ISSUE THREAD

**Issue by petro-i** [fluttercommunity/get_it#196]
Hi please see code sample below:

```dart
GetIt getIt = GetIt.instance;

void main() async {
  getIt.registerSingleton<Service1>(Service1(), signalsReady: true);
  getIt.registerSingletonWithDependencies<Service2>(() => Service2(),
      dependsOn: [Service1], signalsReady: true);

  await getIt.allReady();

  final service2 = getIt<Service2>();

  runApp(MyApp());
}
class Service1 {
  Service1() {
    _init();
  }

  _init() async {
    await Future.delayed(Duration(milliseconds: 1));
    getIt.signalReady(this);
  }
}

class Service2 {
  Service2() {
    _init();
  }

  _init() async {
    await Future.delayed(Duration(milliseconds: 1));
    getIt.signalReady(this);
  }
}`

as result on code `final service2 = getIt<Service2>();` 
exception is reported: `You tried to access an instance of Service2 that is not ready yet`

It is observed that `registerSingletonWithDependencies<Service2>` really awaits until Service1 reports ready, but allReady doesn't await for Service2 reports ready.

GetIt v.7.1.3

**Comment by stx** [comment]
Can confirm. @escamoteur Is there any workaround to this or an ETA on a fix?

**Comment by escamoteur** [comment]
I'm looking into it but it's a bit tricky to debug

**Comment by stx** [comment]
Found a workaround. registerAsync with dependsOn works as you’d expect.

**Comment by escamoteur** [comment]
indeed that's what I would have expected and how most people use it. But the other constellation should work too

**Comment by escamoteur** [comment]
ah, that was indeed a tricky one :-) I found and hopefully fixed in in V7.1.4
